src/main/drivers/stm32/dshot_bitbang.c

```diff
@@ -533,21 +533,13 @@ static bool bbUpdateStart(void)
         if (telemetryWait) {
             DEBUG_SET(DEBUG_DSHOT_TELEMETRY_COUNTS, 2, debug[2] + 1);
         } else {
-            for (int motorIndex = 0; motorIndex < MAX_SUPPORTED_MOTORS && motorIndex < motorCount; motorIndex++) {
 #ifdef USE_DSHOT_CACHE_MGMT
-                // Only invalidate the buffer once. If all motors are on a common port they'll share a buffer.
-                bool invalidated = false;
-                for (int i = 0; i < motorIndex; i++) {
-                    if (bbMotors[motorIndex].bbPort->portInputBuffer == bbMotors[i].bbPort->portInputBuffer) {
-                        invalidated = true;
-                    }
-                }
-                if (!invalidated) {
-                    SCB_InvalidateDCache_by_Addr((uint32_t *)bbMotors[motorIndex].bbPort->portInputBuffer,
-                                                 DSHOT_BB_PORT_IP_BUF_CACHE_ALIGN_BYTES);
-                }
+            for (int i = 0; i < usedMotorPorts; i++) {
+                bbPort_t *bbPort = &bbPorts[i];
+                SCB_InvalidateDCache_by_Addr((uint32_t *)bbPort->portInputBuffer, DSHOT_BB_PORT_IP_BUF_CACHE_ALIGN_BYTES);
+            }
 #endif
-
+            for (int motorIndex = 0; motorIndex < MAX_SUPPORTED_MOTORS && motorIndex < motorCount; motorIndex++) {
```

Comment:
usedMotorPorts represent the validated motors as motorCount is just a mixer attribute for the model.

Reply:
`usedMotorPorts` is the number of ports; not directly connected to the number of  motors. If you have two motors in port A, one on port B and one on port C, for example, then it would equal 3.